### DISCUSSION TITLE
Empty HTML template but CSS is from other pages

### DISCUSSION THREAD

**Discussion by nguyenvoanhkhoa1** [GrapesJS/grapesjs#5414] [Q&A] [answered]
I am developing a multiple-page project editor using pageManager and now is my case: 
The image below is showing an empty HTML template, but a massive CSS code. I found the reason is that there are other components on other pages.
![image](https://github.com/GrapesJS/grapesjs/assets/55738921/89aa7e64-f81a-4b64-88b0-4dd251d5575b)
I am not sure if I use the pageManager right or how to define component styles in the right way. Maybe the CSS selector should be selected by id but not class? I don't know. Anyone can show me a direction or document about it would be grateful.
Thank you for reading.

**Comment by artf** [comment, marked as answer]
Here is the documentation about component-related styles https://grapesjs.com/docs/modules/Components.html#components-css

**Reply by nguyenvoanhkhoa1** [reply]
Hi @artf , I fixed the problem by rewriting the CSS selector. My old selector used tag selector and descendant selector. I turn to BEM class so it does not happen anymore. But there are still some special case like` .my-class:not(:first-child)`, and it appears on all my page if existed 1 component exists on one page.
Thank you for your help.

**Reply by lifeofsa** [reply]
@nguyenvoanhkhoa1 can you please explain how did you do it? I'm stuck in the same issue.?

**Comment by lifeofsa** [comment]
Ok man, thanks got it we just can't style the tag selector directly in css
we have to give an id or className to them for styling... is that right?

**Reply by nguyenvoanhkhoa1** [reply]
one more important thing is in this doc [https://grapesjs.com/docs/modules/Blocks.html#important-caveats](url)

**Comment by lifeofsa** [comment]
Got it, thanks!
